File: Lista 4/Java/Contato.java
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class Contato {
    private String nome;
    private String email;
    private String telefone;
    private LocalDate dataNascimento;

    // Inicializar contato
    public void inicializarContato(String nome, String email, String telefone, LocalDate dataNascimento){
        this.nome = nome;
        this.email = email;
        this.telefone = telefone;
        this.dataNascimento = dataNascimento;
    }

    // Imprimir contato
    public void imprimirContato(){
        System.out.println("Nome: " + nome);
        System.out.println("Email: " + email);
        System.out.println("Telefone: " + telefone);
        System.out.println("Data de Nascimento: " + DateTimeFormatter.ofPattern("dd/MM/yyyy").format(dataNascimento));
    }

    // Calcular idade
    public int calcularIdade(){
        LocalDate dataHoje = LocalDate.now();
        int idade = dataHoje.getYear() - dataNascimento.getYear();

        if (dataHoje.getMonthValue() < dataNascimento.getMonthValue()
                || (dataHoje.getMonthValue() == dataNascimento.getMonthValue() && dataHoje.getDayOfMonth() < dataNascimento.getDayOfMonth()))
            idade--;

        return idade;
    }

    // Getters
    public String getNome(){
        return nome;
    }

    public String getEmail(){
        return email;
    }

    public String getTelefone(){
        return telefone;
    }
    
    public LocalDate getDataNascimento(){
        return dataNascimento;
    }

    // Setters
    public void setNome(String nome){
        this.nome = nome;
    }

    public void setEmail(String email){
        this.email = email;
    }

    public void setTelefone(String telefone){
        this.telefone = telefone;
    }

    public void setDataNascimento(LocalDate dataNascimento){
        this.dataNascimento = dataNascimento;
    }


    public static void main(String[] args){
        Contato contato = new Contato();
        contato.inicializarContato("Fulano", "fulano@email.com", "73999999999", LocalDate.of(2000, 5, 25));
        contato.imprimirContato();

        contato.setNome("Ciclano");
        contato.setEmail("ciclano@email.com");
        contato.setTelefone("73888888888");
        
        System.out.printf("%s, %s, %s, %s", contato.getNome(), contato.getEmail(), contato.getTelefone(), contato.getDataNascimento());
        System.out.println(contato.calcularIdade());
    }
    
}

File: Lista 4/Python/Contato.py
from datetime import datetime as dt

class Contato:
    def __init__(self, nome = "", email = "", telefone = "", data_nascimento = ""):
        self.__nome = nome
        self.__email = email
        self.__telefone = telefone
        self.__data_nascimento = data_nascimento

    # Imprimir contato
    def __str__(self):
        return f"Nome: {self.__nome}\n" \
               f"Email: {self.__email}\n" \
               f"Telefone: {self.__telefone}\n" \
               f"Data de Nascimento: {self.__data_nascimento}"

    # Calcular idade
    def calcular_idade(self):
        diferenca = dt.today() - dt.strptime(self.__data_nascimento, "%d/%m/%Y")
        idade = int(diferenca.days / 365.25)
        return idade

    # Getters
    @property
    def nome(self):
        return self.__nome
    
    @property
    def email(self):
        return self.__email
    
    @property
    def telefone(self):
        return self.__telefone
    
    @property
    def data_nascimento(self):
        return self.__data_nascimento

    # Setters
    @nome.setter
    def nome(self, nome):
        self.__nome = nome

    @email.setter
    def email(self, email):
        self.__email = email

    @telefone.setter
    def telefone(self, telefone):
        self.__telefone = telefone

    @data_nascimento.setter
    def data_nascimento(self, data_nascimento):
        try:
            dt.strptime(data_nascimento, "%d/%m/%Y")
        except ValueError:
            raise ValueError("Data de nascimento deve estar no formato dd/mm/aaaa.")
        self.__data_nascimento = data_nascimento


if __name__ == "__main__":
    contato = Contato("Fulano", "fulano@email.com", "73999999999", "26/04/2000")
    print(contato)
    print("Idade", contato.calcular_idade())

    contato.nome = "Ciclano"
    contato.email = "ciclano@email.com"
    contato.telefone = "73888888888"
    contato.data_nascimento = "26/04/2000"
    print(contato)
    print("Idade:", contato.calcular_idade())